integration-tests/src/androidTest/java/com/dropbox/kaiken/skeleton/scoping/ScopesConstructionTests.kt

```diff
@@ -0,0 +1,122 @@
+package com.dropbox.kaiken.skeleton.scoping
+
+
+import com.dropbox.kaiken.skeleton.core.AppSpecificScopedServicesFactory
+import com.dropbox.kaiken.skeleton.core.KaikenConfig
+import com.dropbox.kaiken.skeleton.core.SkeletonAccessTokenPair
+import com.dropbox.kaiken.skeleton.core.SkeletonConfig
+import com.dropbox.kaiken.skeleton.core.SkeletonUser
+import com.dropbox.kaiken.skeleton.dagger.SdkSpec
+import com.dropbox.kaiken.skeleton.usermanagement.auth.ActiveUserManager
+import com.dropbox.kaiken.skeleton.usermanagement.auth.RealActiveUserManager
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesTo
+import com.squareup.anvil.annotations.MergeComponent
+import dagger.Component
+import org.junit.Test
+import javax.inject.Inject
+
+
+class ScopesConstructionTests {
+    @Test
+    fun constructScopes() {
+        val skeletonServics = DaggerSkeletonTestComponent.factory().create()
+        val appServices: AppServices =
+            skeletonServics.getSkeletonConfig().scopedServicesFactory.createAppServices(
+                skeletonServics
+            )
+        val userServices: UserServices =
+            skeletonServics.getSkeletonConfig().scopedServicesFactory.createUserServices(
+                appServices,
+                SkeletonUser("1", SkeletonAccessTokenPair("1", "1"), isActiveUser = true)
+            )
+        val authActivityComponent: AuthRequiredActivityComponent =
+            userServices.cast<AuthRequiredActivityComponent.ParentComponent>()
+                .createAuthRequiredComponent()
+        val noAuthActivityComponent: AuthOptionalActivityComponent =
+            appServices.cast<AuthOptionalActivityComponent.ParentComponent>()
+                .createAuthOptionalComponent()
+        assertThat(authActivityComponent.cast<UserTeardownHelperProvider>().userTeardownHelper()).isInstanceOf(UserTeardownHelper::class.java)
+        assertThat(noAuthActivityComponent.cast<AppTeardownHelperProvider>().appTeardownHelper()).isInstanceOf(AppTeardownHelper::class.java)
+    }
+}
+
+@MergeComponent(SkeletonScope::class)
+@SingleIn(SkeletonScope::class)
+interface SkeletonTestComponent : SdkSpec {
+    @Component.Factory
+    interface Factory {
+        fun create(
+//            @BindsInstance app: SkeletonOwner,
+        ): SkeletonTestComponent
+    }
+}
+
+@ContributesBinding(SkeletonScope::class, replaces = [KaikenConfig::class])
+class FakeSkeletonConfig @Inject constructor() : SkeletonConfig {
```

Comment:
I looooove anvil. We have a binding that is contributed somewhere within skeleton main source. In tests, I can change this binding. One of the ways I can change it is by making a fake SkeletonConfig which no longer depends on an application class. Next, I can make the `SkeletonTestComponent` which never binds the application instance in. This way you can have variants of the skeleton graph which are decoupled from android. 

Similar check out the `FakeActiveUserManager` below which drops the real dependency on coroutines for this noop version